forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/FormBPMDataPipelineListener.java

```diff
@@ -0,0 +1,53 @@
+package org.camunda.bpm.extension.hooks.listeners;
+
+
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.delegate.DelegateTask;
+import org.camunda.bpm.engine.delegate.ExecutionListener;
+import org.camunda.bpm.engine.delegate.TaskListener;
+import org.camunda.bpm.extension.hooks.services.FormSubmissionService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.inject.Named;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * This class transforms all the form document data into CAM variables
+ *
+ * @author sumathi.thirumani@aot-technologies.com
+ */
+@Named("FormBPMDataPipelineListener")
+public class FormBPMDataPipelineListener  extends BaseListener implements TaskListener, ExecutionListener {
+
+    @Autowired
+    private FormSubmissionService formSubmissionService;
+
+    @Override
+    public void notify(DelegateExecution execution) {
+        try {
+            syncFormVariables(execution);
+        } catch (IOException e) {
+            handleException(execution, ExceptionSource.EXECUTION, e);
+        }
+    }
+
+    @Override
+    public void notify(DelegateTask delegateTask) {
+        try {
+            syncFormVariables(delegateTask.getExecution());
+        } catch (IOException e) {
+            handleException(delegateTask.getExecution(), ExceptionSource.TASK, e);
+        }
+    }
+
+    private void syncFormVariables(DelegateExecution execution) throws IOException {
+        Map<String,Object> dataMap = formSubmissionService.retrieveFormValues(String.valueOf(execution.getVariables().get("formUrl")));
+        for (Map.Entry<String, Object> entry: dataMap.entrySet()) {
+            String fieldValue = String.valueOf(entry.getValue());
+            if (fieldValue.length() < 4000) {
+                execution.setVariable(entry.getKey(), entry.getValue());
+            }
```

Comment:
We need to add this limitation to the document. I suggest if it's not difficult or in another ticket, let the user know of this limitation on UI (add a label or something).

Reply:
Probably the one @simensma-fresh mentioned on how to use Camunda. Maybe in the Camunda readme?